addons/binding/org.openhab.binding.tellstick/src/main/java/org/openhab/binding/tellstick/internal/core/TelldusCoreBridgeHandler.java

```diff
@@ -191,6 +201,8 @@ public boolean unregisterDeviceStatusListener(DeviceStatusListener deviceStatusL
     }
 
     public void clearDeviceList() {
+        deviceList.clear();
```

Comment:
Its not clear to me where this comes from and why it was not like this before? The rest looks fine to me.

Reply:
Well, it comes from the fact that the dispose method should leave the handler instance in a state as if it was newly created - so clearing the lists is definitely something required here.

I am not sure though whether anything further needs to be done with the list entries (unregistering listeners, etc). But that isn't yet done anyhow, so this PR should only make things better, but not worse. I am not saying that it makes things fully perfect though.

Reply:
👍

Reply:
The devicelist is actually recreated every time so this should not make any difference, the null checks are obviously needed.

Reply:
The only addon I see is
if (eventHandler != null) {
            eventHandler.removeListener((TelldusCoreDeviceController) deviceController);
            eventHandler.removeListener(this);
            eventHandler.remove();
            eventHandler = null;
        }

Reply:
> The devicelist is actually recreated every time

I only see it created once in the constructor, but never again on initialize. Therefore I think that clearing it on dispose is required.

Reply:
@jarlebh are you aware that handler's are 'recycled' / refreshed by calling dispose and reinitializing after that. So dispose it not only used as a destructor but also to re-initialize.

Reply:
I have noticed that devices that I have removed from my inbox keep resurfacing from this binding, don't know if that's related to how this is done?

Reply:
Not sure about the inbox thing, when should the binding report an thing. Only this first time it sees the thing ?
I was not aware of the recycling, I assumed it created a new Handler.

Reply:
The deviceList is actually recreated for every call to rescan, so this change should not make any difference: 
 deviceList = Collections.synchronizedList(TellstickDevice.getDevices());

Reply:
Agree on that, given its a little bit to save I would say that we can give @kaikreuzer the green light